daemon/src/taker_cfd.rs

```diff
@@ -228,12 +232,62 @@ impl Actor {
         self.cfd_feed_actor_inbox
             .send(load_all_cfds(&mut conn).await?)?;
 
-        let txid = self.wallet.try_broadcast_transaction(dlc.lock.0).await?;
+        let txid = self
+            .wallet
+            .try_broadcast_transaction(dlc.lock.0.clone())
+            .await?;
 
         tracing::info!("Lock transaction published with txid {}", txid);
 
-        // TODO: tx monitoring, once confirmed with x blocks transition the Cfd to
-        // Open
+        // TODO: It's a bit suspicious to load this just to get the
+        // refund timelock
+        let cfd = load_cfd_by_order_id(order_id, &mut conn).await?;
+
+        let script_pubkey = dlc.address.script_pubkey();
+        self.monitor_actor
+            .do_send_async(monitor::StartMonitoring {
+                id: order_id,
+                params: MonitorParams {
+                    lock: (dlc.lock.0.txid(), dlc.lock.1),
+                    commit: (dlc.commit.0.txid(), dlc.commit.2),
+                    cets: dlc
+                        .cets
+                        .into_iter()
+                        .map(|(tx, _, range)| (tx.txid(), script_pubkey.clone(), range))
+                        .collect(),
+                    refund: (
+                        dlc.refund.0.txid(),
+                        script_pubkey,
+                        cfd.refund_timelock_in_blocks(),
+                    ),
+                },
+            })
+            .await?;
+
+        Ok(())
+    }
+
+    async fn handle_monitoring_event(&mut self, event: CfdMonitoringEvent) -> Result<()> {
+        let order_id = event.order_id();
+
+        let mut conn = self.db.acquire().await?;
+        let cfd = load_cfd_by_order_id(order_id, &mut conn).await?;
+
+        let new_state = cfd.transition_to(CfdStateChangeEvent::Monitor(event))?;
+
+        insert_new_cfd_state_by_order_id(order_id, new_state.clone(), &mut conn).await?;
+
+        // TODO: Not sure that should be done here...
+        //  Consider sending a message to ourselves to trigger broadcasting refund?
+        if let CfdState::MustRefund { .. } = new_state {
+            let signed_refund_tx = cfd.refund_tx()?;
+            let txid = self
+                .wallet
+                .try_broadcast_transaction(signed_refund_tx)
+                .await?;
+
+            tracing::info!("Refund transaction published on chain: {}", txid);
+        }
```

Comment:
Didn't we want triggering of these things up to the user?

Reply:
When implementing this with @luckysori we went by the convention: If the user has a choice we bubble it up, if the user does not have a choice we act. Refund is not choice. To my understanding, if the refund timelock expired you have to refund. Or is there a scenario where it is save to do anything but refund then? 

We can still give the user a choice based on the state and only trigger the publication upon user request. Shall I record it as a ticket?

Reply:
The assumption is that you only reach this state if the oracle never attests a price and the refund timelock has expired. At this stage I don't think the application should wait for anything: it should refund automatically.